### PR TITLE
Migrate to newer `ProcessManager` API

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR migrates to newer `core-java` which recently introduced `ProcessManager.select()` method.

Also `Renderer.protoDataContext` `var` was made private in favor of new `@Internal` method `injectContext()`.

### CODE REVIEW

File: api/src/main/kotlin/io/spine/protodata/renderer/Renderer.kt [hunk 1]
```diff
@@ -74,4 +75,21 @@ protected constructor(
...
+     *
+     * This method is `public` but is essentially `internal` to ProtoData SDK.
+     * It is annotated `@Internal` to prevent its accidental use by the users of the SDK.
+     *
+     * This method must be called by ProtoData only once.
+     * Repeat calls will result in `IllegalStateException`.
+     */
+    @Internal
+    public fun injectContext(protoDataContext: BoundedContext) {
+        check(!this::protoDataContext.isInitialized) {
```
Comment (dmdashenkov) [inline api/src/main/kotlin/io/spine/protodata/renderer/Renderer.kt:90]: Please use the `ContextAware` interface along with the check methods provided there.

**Review by dmdashenkov — CHANGES_REQUESTED** [review]
Please see my comment.

File: api/src/main/kotlin/io/spine/protodata/renderer/Renderer.kt [hunk 2]
```diff
@@ -74,4 +75,28 @@ protected constructor(
...
+    public override fun registerWith(protoDataContext: BoundedContext) {
+        if (isRegistered) {
+            check(this.protoDataContext == protoDataContext) {
+                "Unable to register the renderer `$this` with" +
+                        " `${protoDataContext.name().value}`." +
+                        " The renderer is already registered with" +
+                        " `${this.protoDataContext.name().value}`."
+            }
+            return
+        }
```
Comment (dmdashenkov) [inline api/src/main/kotlin/io/spine/protodata/renderer/Renderer.kt:95]: Please use `checkNotRegistered()` instead.

**Review by dmdashenkov — APPROVED** [review]
LGTM with a single suggestion.

File: api/src/main/kotlin/io/spine/protodata/renderer/Renderer.kt (hunk 2 above)
Comment (alexander-yevsyukov) [reply api/src/main/kotlin/io/spine/protodata/renderer/Renderer.kt:95]: I thought of this, but `checkNotRegistered()` does not give enough diagnostics. It simply tells that an object is already registered. We need to tell a bit more in such a delicate case.

**Pull request merged**